### PR TITLE
Builders Get contract

### LINKED ISSUE #448: Builders Get contract
To align with user intent and make the Builder discoverable, the methods in the Container to access the Change Feed Builders need to be changed to:

* `GetChangeFeedProcessorBuilder`
* `GetChangeFeedEstimatorBuilder`

For Container, changing `DefineContainer` to `GetContainerBuilder` and `CreateContainerDefinition` to `ContainerBuilder`.

### PR DESCRIPTION
**Pull request by ealsur**
# Builders Get contract

## Description

To align with user intent and make the Builder discoverable, the methods in the Container to access the Change Feed Builders are changed to:

* `GetChangeFeedProcessorBuilder`
* `GetChangeFeedEstimatorBuilder`

For Container, changing `DefineContainer` to `GetContainerBuilder` and `CreateContainerDefinition` to `ContainerBuilder`.

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

Closes #448

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
Database.DefineContainer -> Database.GetContainerBuilder -> ContainerBuilder (TypeRename).

Will it also be covered?

**Comment by bartelink** [comment]
Suggest `GetChangeFeedProcessorBuilder` or `BuildChangeFeedProcessor` - you're not `Get`ing anything, you're _creating_ it

**Comment by kirankumarkolli** [comment]
ChangeFeedProcessorBuilder is returning the builder only not the CFProcessor.
Create verb might mis-lead hence Get.

**Comment by bartelink** [comment]
@kirankumarkolli Apologies - I mistyped when I made my comment - I meant (and have edited in) `CreateChangeFeedProcessorBuilder`

My reasoning:

- `GetChangeFeedProcessorBuilder` is _Creating_ a new Builder - it is not Getting _the_ Builder. Hence I really don't feel this is a good name; there is no prior art for this kind of naming, and `Get`, like `Manager` etc. is on the FDG "bad verbs list" in my head
- The word `Build` or `Builder` should definitely be in the name somewhere. If there's going to be a verb other than `Build`, it definitely should not be `Get`; it should be `Create`.

In conclusion, I'd suggest, in order of preference:
1. `BuildChangeFeedProcessor` (mentions builder pattern, suggests we are building something new)
2. `CreateChangeFeedProcessorBuilder` (long, but descriptive of what its doing using well established verbs per FDG and general Pattern names)
3. `GetChangeFeedProcessorBuilder` (you're not getting _the_ builder, and there's no implication that we're starting a chain of building a thing)
